ggml/src/ggml-cuda/softmax.cu

```diff
@@ -150,6 +151,24 @@ static __global__ void soft_max_back_f32(
     }
 }
 
+template<int... Ns>
+void increase_shared_mem_limits(std::size_t smpbo)
+{
+    auto apply_limit = [smpbo](auto I) {
+        constexpr int ncols  = decltype(I)::value;
+        constexpr int block  = (ncols > 1024 ? 1024 : ncols);
+
+        CUDA_SET_SHARED_MEMORY_LIMIT(
+            (soft_max_f32<true, ncols, block, half >),  smpbo);
+        CUDA_SET_SHARED_MEMORY_LIMIT(
+            (soft_max_f32<true, ncols, block, float>), smpbo);
+    };
+
+    //unary fold
+    ( apply_limit(std::integral_constant<int, Ns>{}), ... );
+}
```

Comment:
Define a macro for a switch case instead? I would (slightly) prefer it to avoid duplicating the logic for setting the block size as a function of `ncols`. I think this would ultimately also be fine though.